x-pack/test/api_integration/apis/search/session.ts

```diff
@@ -428,17 +426,21 @@ export default function ({ getService }: FtrProviderContext) {
       // it might take the session a moment to be updated
       await new Promise((resolve) => setTimeout(resolve, 2500));
 
-      const getSessionSecondTime = await supertest
-        .get(`/internal/session/${sessionId}`)
-        .set('kbn-xsrf', 'foo')
-        .expect(200);
+      await retry.waitFor('search session touched time updated', async () => {
+        const getSessionSecondTime = await supertest
```

Comment:
This is specifically where it failed in https://github.com/elastic/kibana/issues/119660

Reply:
Can we remove the `await new Promise((resolve) => setTimeout(resolve, 2500));` two lines above this? Or do we risk hitting the timeout in that case? (And if so, should we consider using `waitForWithTimeout` here instead?)

Reply:
We just know that it will take at the very minimum 2.5s. So we wait just to not poll es during that time

Reply:
I guess I'm not sure as to why it will always take 2.5s. Is it because of `trackingInterval`? Because isn't that a separate task that is independent of these tests?

Reply:
Ah, this isn't updated during monitoring task, but updated with `trackId()` call during search request, but without waiting for trackId to complete. 

trackId is scheduled with 1s debounce up to 5s max: 

https://github.com/elastic/kibana/blob/d40c0abe5bc1d0a1c3514068603f0f9a0dffc7a7/x-pack/plugins/data_enhanced/server/search/session/session_service.ts#L464


I guess 2.5s seconds here is a decent guess, but we indeed can remove it and just allow tests to check more

Reply:
update may also take longer because of conflict resolution logic and retries

Reply:
I think it fair to keep that wait, as it is expected that updating session will take some time